hazelcast-client/src/main/java/com/hazelcast/client/cache/impl/NearCachedClientCacheProxy.java

```diff
@@ -349,11 +349,19 @@ public void clear() {
     }
 
     @Override
-    protected Object invokeInternal(Data keyData, Data epData, Object[] arguments) {
+    protected Data invokeInternal(Data keyData, Data epData, Object[] arguments) {
+        Data valueData = null;
         try {
-            return super.invokeInternal(keyData, epData, arguments);
+            valueData = super.invokeInternal(keyData, epData, arguments);
+            return valueData;
         } finally {
-            invalidateNearCache(keyData);
+            // we just have the value as `Data` here, so it doesn't make sense to use cacheOrInvalidate(),
+            // which looks for the best candidate between `Object` and `Data` format
+            if (cacheOnUpdate && valueData != null) {
```

Comment:
I think we can get `null` here, but I'm still missing that test case. I just faked it an got an assertion error later, since we are not allowed to store `null`. So I'm sure that we are not allowed to safe `null` here. But I'm not 100% sure if it's correct to call `invalidateNearCache(keyData);` in that case or if I should just ignore the result.